lang/sbcl/Portfile

```diff
@@ -132,24 +132,18 @@ build.target    make.sh
 build.args-append \
                 --prefix=${prefix}
 
-# unfortunately sbcl-bootstrap doesn't support macOS Sonoma yet, use ECL instead
-# See: https://trac.macports.org/ticket/68271
-if {${os.major} >= 23} {
-    depends_build-append \
-                path:bin/ecl:ecl
+variant bootstrap_ecl description { Use ECL to bootstrap compilation instead of SBCL binary. } {}
 
-    build.args-append \
-                --xc-host=${prefix}/bin/ecl
+if { ![variant_isset bootstrap_ecl]  } {
```

Comment:
This may need to be restricted to archs where it actually works. I think it is broken on i386, ppc and ppc64 (or at least untested on the latter).
@catap will know better, perhaps.

Reply:
I don't have the hardware to test anywhere other than arm64-macos14, so I can't contribute to verification.

Reply:
Your proposal to restrict the `bootstrap_ecl` makes sense but I am not quite sure how to implement.  Is there a canonical way to restrict variants by architecture, or can you provide an example from some Portfile that I might crib?

Reply:
@barracuda156 this was made as hack to allow build SBCL on macOS 14. After they had released the nice version, we don't need it.

So, I suggest to keep it as variant if it will be needed again.

Reply:
> Is there a canonical way to restrict variants by architecture

Not that I know of. I can't recall this need arising before. You can add a `pre-configure` block in the variant that checks the architecture and use `return -code error "some explanation"` if the conditions aren't right.

Reply:
What stops us from wrapping a variant inside a check for build_arch? I think I have seen it in other ports.

Having said that, not a big deal. If someone gonna complain on Trac, it can be addressed later.

Reply:
> What stops us from wrapping a variant inside a check for build_arch? I think I have seen it in other ports.

Variants should not exist conditionally. This is bad for the portindexing process and the distfile mirroring process.